src/wmtk/operations/tet_mesh/EdgeOperationData.hpp

```diff
@@ -148,5 +155,12 @@ class EdgeOperationData : public wmtk::operations::EdgeOperationData
     std::vector<TetCollapseData> tet_collapse_data;
     std::vector<FaceCollapseData> face_collapse_data;
     std::vector<TetSplitData> tet_split_data;
+
+    // data required for swap
+    // TODO: can be used as cache for speedup
+    std::vector<long> m_new_tet_ids;
```

Comment:
there is a lot more structure to this than represented here. i would rather store these values according to their structure and then reconstruct htis when necessary later rather than just lose the structure.
also if you're going to create 4 vectors like this just do an array of vectosr anyway